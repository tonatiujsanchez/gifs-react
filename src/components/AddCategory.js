import { useState } from "react"
import PropTypes from 'prop-types'

const AddCategory = ({ addNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('')


    const handleChange = ( e ) => {
        setInputValue( e.target.value )
    }


    const handleSubmit = (e) =>{
        e.preventDefault()
        
        if( inputValue.trim() !== '' ){
            addNewCategory( inputValue )
            setInputValue('')        
        }
        
    }


  return (
    <form onSubmit={ handleSubmit }>
        <p>{ inputValue }</p>
        <input 
            type="text" 
            value={ inputValue } 
            onChange={ handleChange }
            placeholder= 'One Punch, Samurai X, Dragon Ball, Etc'/>
    </form>
  )
}

AddCategory.propTypes ={
    addNewCategory: PropTypes.func.isRequired
}

export default AddCategory