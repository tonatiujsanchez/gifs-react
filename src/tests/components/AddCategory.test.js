import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"




describe('Priebas en el Componente AddCategory', () => { 
    
    const addNewCategory = jest.fn()
    let wrapper = shallow( <AddCategory addNewCategory={ addNewCategory } /> )

    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallow( <AddCategory addNewCategory={ addNewCategory } /> )
    })


    test('Debe de mostrarse correctamente', () => { 

        expect( wrapper ).toMatchSnapshot()

     })


     test('Debe de cambiar la caja de texto', () => { 


        const input = wrapper.find('input')

        const value = 'Hola mundo'
        input.simulate('change', { target: { value } })


        const parrafo = wrapper.find('p').text().trim()

        expect( parrafo ).toBe( value )
    
      })



      test('No debe se enviar la informacion onSubmit', () => { 

        const formulario = wrapper.find('form')

        formulario.simulate('submit', { preventDefault(){}  })


        expect( addNewCategory ).not.toHaveBeenCalled()

       })


       test('Debe de llamr addNewCategory y limpiar la caja de Texto', () => { 
           
           let value = 'Hello word'

            // 1. Simulación del inputChange
            const input = wrapper.find('input')
            input.simulate('change', { target:{ value } })

            // 2. Simulación del onSubmit
            const formulario = wrapper.find('form' )
            formulario.simulate('submit', {  preventDefault(){} })


            // 3. Verificar que addNewCategory() se llamó por lo menos una vez
            expect( addNewCategory ).toHaveBeenCalled()

            // 3.1. Verificar que addNewCategory() se llamó n cantidad de veces
            // expect( addNewCategory ).toHaveBeenCalledTimes(2) 
            
            // 3.2. Verificar que addNewCategory() se llamó y se envio un String
            expect( addNewCategory ).toHaveBeenCalledWith( expect.any(String) )

                    
            //4. Verificar que el input se limpió
            expect( input.prop('value') ).toBe('')

        })




 })