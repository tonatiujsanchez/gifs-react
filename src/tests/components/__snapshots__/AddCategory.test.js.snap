// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Priebas en el Componente AddCategory Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="One Punch, Samurai X, Dragon Ball, Etc"
    type="text"
    value=""
  />
</form>
`;
